homeassistant/components/vallox/config_flow.py

```diff
@@ -0,0 +1,120 @@
+"""Config flow for the Vallox integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from vallox_websocket_api import Vallox
+from vallox_websocket_api.exceptions import ValloxApiException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.util.network import is_ip_address
+
+from .const import DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+VALLOX_CONNECTION_EXCEPTIONS = (
+    OSError,
+    ValloxApiException,
+)
+
+
+async def validate_host(hass: HomeAssistant, host: str) -> None:
+    """Validate that the user input allows us to connect."""
+
+    if not is_ip_address(host):
+        raise InvalidHost(f"Invalid IP address: {host}")
+
+    client = Vallox(host)
+    await client.get_info()
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the Vallox integration."""
+
+    VERSION = 1
+
+    async def async_step_import(self, data: dict[str, Any]) -> FlowResult:
+        """Handle import from YAML."""
+        # We need to use the name from the YAML configuration to avoid
+        # breaking existing entity IDs.
+        name = data.get(CONF_NAME, DEFAULT_NAME)
+        host = data[CONF_HOST]
+
+        self._async_abort_entries_match({CONF_HOST: host})
+
+        reason = None
+        try:
+            await validate_host(self.hass, host)
+        except InvalidHost:
+            _LOGGER.exception("An invalid host is configured for Vallox")
```

Comment:
We normally only log the stack trace for unknown exceptions.

Reply:
Thanks, fixed this and all other comments in https://github.com/home-assistant/core/pull/63032.